### DISCUSSION TITLE
Setting up github copilot on gitlab hosted repositories and pipelines.

### DISCUSSION THREAD

**Discussion by steve-kuelap** [community/community#66976] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Hello, 

We are trying to setup Github copilot for company wide repositories hosted on gitlab. Some of the intended use cases include the ability to have preliminary code reviews and code suggestions when commits are pushed to the gitlab repositories.

However, I don't seem to find if that will be a possibility given that the code is hosted on gitlab and not github.

**Comment by thanmaisai** [comment]
GitHub Copilot is currently only available for repositories hosted on GitHub.com. However, GitHub is working on a feature called Copilot X that will allow GitHub Copilot to be used with repositories hosted on other platforms, such as GitLab.

Copilot X is still in development, and there is no ETA for its release. However, if you are interested in using GitHub Copilot with your company's GitLab repositories, you can sign up for the Copilot X waitlist.

you can use other tools to get code suggestions and preliminary code reviews for your GitLab repositories. 
Some popular tools include CodeFactor, Codacy, PullReview.

**Comment by seanmac5291** [comment, marked as answer]
@steve-kuelap @thanmaisai 

You absolutely can use GitHub Copilot Business  for code completion and chat capabilities regardless of where your code lives. It is available as an extension to the IDE (Visual Studio Code, Visual Studio, jetbrains, and a couple others), so location of code doesn't matter. 
 
https://docs.github.com/en/enterprise-cloud@latest/copilot/copilot-business/about-github-copilot-business#about-copilot-business
 
Copilot Enterprise  - was released on Feb 27th and does require all code to live in GitHub Enterprise, as the AI capabilities are infused into the GHE platform in order to assist with things like writing pull request summaries, and the chat capability right in their enterprise dashboard, and more to come.
 
https://github.blog/2024-02-27-github-copilot-enterprise-is-now-generally-available/
 
Copilot X itself is actually more of a vision than a product per se, and speaks to where GitHub is ultimately taking GH Copilot and how we plan to use AI to help development teams write secure code more efficiently.
 
https://github.blog/2023-03-22-github-copilot-x-the-ai-powered-developer-experience/
